latex2sympy.py

```diff
@@ -104,16 +104,37 @@ def process_sympy(self):
     def pre_process_latex(self, latex):
         '''
         pre-processing for issues the parser cannot handle
+
+        find any single char sup/sub and wrap them in "{}"
+        e.g. `4^26^2` => `4^{2}6^{2}`
+        e.g. `x_22` => `x_{2}2`
+
+        NOTE: this DOES MODIFY variable names so they can be parsed,
+        but `atom.VARIABLE()` reverts to the original name, if needed, using `self.variable_name_dict`
+
         '''
 
-        # find any single char sup/sub and wrap them in "{}"
-        # e.g. `4^26^2` => `4^{2}6^{2}`
-        # e.g. `x_22` => `x_{2}2`
+        # pattern to find variable commands
+        # match groups:
+        # 0: the variable name up to the first closing "}". could be the "}" for the subscript, or the variable itself (see group 3)
+        # 1: the entire subscript (optional)
+        # 2: the opening subscript bracket (optional)
+        # 3: a second closing "}" (optional). could be from a wrapping command. IF group 2 is NOT "None", this is the closing subscript "}".
+        variable_regex = r'\\variable{([^}_]+(_({)?[^}]+)?)}(})?'
```

Comment:
@builtbywill This looks great. The only comment I have is that this regex could be updated a bit to capture the whole variable name so you don't need the string join below. The regex below will grab the whole variable name and also won't match on a couple invalid variable names. I have a link with some examples too.

https://regex101.com/r/z6bglU/2
`\\variable{([^}_]+?(_({[^}_]+?}|[^}_]))?)}`

Reply:
@tragessere thanks! that looks great, added it in